### ISSUE TITLE
[v2] markup different client and server

### ISSUE THREAD

**Issue by pungggi** [styled-components/styled-components#513]
## Version
2.0.0-4

I get 
Warning: React attempted to reuse markup in a container but the checksum was invalid. This generally means that you are using server rendering and the markup generated on the server was not what the client was expecting. React injected new markup to compensate which works but you have lost many of the benefits of server rendering. Instead, figure out why the markup being generated is different on the client or server:
 (client) ="1"><div class="sc-gqjmRU cFUSVC" data-
 (server) ="1"><div class="sc-hqyNC jPUbry" data-r

is this something i must investigate more or is this from v2? 
i did not have this in 1.4.3

**Comment by nicksrandall** [comment]
Are you using https://github.com/styled-components/babel-plugin-styled-components? I believe it is still a WIP but it should ensure that you get deterministic class names for your components.

**Comment by pungggi** [comment]
Work fine thank you, i still get that message, but anyway good to have "speaking classes" now I see clearly wich Element is causing it.
![image](https://cloud.githubusercontent.com/assets/5635709/23093463/c6e0703e-f5e4-11e6-8fdb-3ebd37a49a60.png)

Warning: React attempted to reuse markup in a container but the checksum was invalid. This generally means that you are using server rendering and the markup generated on the server was not what the client was expecting. React injected new markup to compensate which works but you have lost many of the benefits of server rendering. Instead, figure out why the markup being generated is different on the client or server:
 (client) ss="page__Container-bodknx eXHyXF" data-
 (server) ss="page__Container-1tu8z7d fRbQnJ" data

**Comment by rovansteen** [comment]
I'm having this issue too, it seems like the class names are generated in a different way on the client and the server?

**Comment by mxstbr** [comment]
Are you running the babel plugin on both the client and the server?

**Comment by rovansteen** [comment]
@mxstbr yes, the human friendly class names appear both on the client and the server generated HTML.

```
 (client) ead__StyledMasthead-1qcthel fGzUif masth
 (server) ead__StyledMasthead-tu6v5q BzWWz masthea
```

**Comment by davidrossjones** [comment]
I'm also experiencing this issue when using SSR with next.js + styled-components v2.0.0-4 regardless of whether I use the babel plugin or not? No such issue with same code and 1.4.

~~~
Instead, figure out why the markup being generated is different on the client or server:
 (client) class="A__A-11k92nv bvKGfC" data-reactid
 (server) class="A__A-11k92nv cDagqo" data-reactid
~~~

**Comment by hampusohlsson** [comment]
Experiencing the same issue as above when using `babel-plugin-styled-components`.

However, the issue does appear slightly different if I remove the plugin - the error only shows up if I hot-reload some component and then refresh the browser (without restarting the server). If I restart the server then error disappears.

Btw @mxstbr, awesome work with styled components. Big fan!

**Comment by jameslnewell** [comment]
The unpublished commit https://github.com/styled-components/babel-plugin-styled-components/commit/e0b1b4c248f4864c65a1fd368bfc71b36d7407c9 fixes this issue for me. @mxstbr @geelen any reason it wasn't published yet?

**Comment by mrmartineau** [comment]
FYI @mxstbr, I have this issue with `"preprocess": true` using the babel plugin (v1.5.0). I know it is experimental, so setting the value to `false` fixes the issue. I'm using next.js.

This is my .babelrc:
```json
{
  "presets": ["next/babel"],
  "plugins": [
    [
      "styled-components",
      { "ssr": true, "displayName": false, "preprocess": false }
    ]
  ]
}
```

**Comment by mxstbr** [comment]
Unfortunately we hit a wall with `preprocess` and that will most likely never be stable/usable. We have some further [plans](https://github.com/cssinjs/istf-spec) and [experiments](https://github.com/kitten/sweetsour) in this direction, but they won't have anything to do with the Babel plugin.

I'd recommend not using that option as it's unfinished and most likely very broken. We'll probably remove it in a future major version.
